### ISSUE TITLE
Revisions for extension's versions

### ISSUE THREAD

**Issue by vmaksimo** [KhronosGroup/SPIRV-LLVM-Translator#1856]
The idea appeared from development and review of #1835

Sometimes the extension specification introduces backward-incompatible changes in its new revision. If we just update the existing implementation to the such new revision, it will affect SPIR-V consumers which are not yet ready for such changes.

The suggestion is to introduce mechanism that will allow to set the certain revision for the extension, if there is more than one supported by translator. Obviously, we won't support every single revision, but a "bunch" of them at once when drastic changes have happened. The translation for different revisions will be done in accordance to the spec, and by default we'll take the "lowest" revision available for extension if it's not set specifically.

The usage example:
```
--spirv-ext=+[EXT_NAME]:[EXT_VERSION]
// e.g.: 
--spirv-ext=+SPV_INTEL_tensor_float32_conversion:5
```

To sum up, this will allow to introduce backward-incompatible changes to the SPIR-V consumers more smoothly.
Want to get feedback on how helpful the improvement looks, and is it useful enough to be implemented.
